File: ioi/2013/day0-1/birthday/grader.c
#include <stdio.h>
#include <stdlib.h>
#include "birthday.h"

#define fail(s, x...) do { \
  fprintf(stderr, s "\n", ## x); \
  exit(1); \
} while(0)

#define MAX_R 500
#define MAX_N 1000

int X1[MAX_N];
int Y1[MAX_N];
int X2[MAX_N];
int Y2[MAX_N];

int main() {
  int R, N, i;
  int res;

  FILE *f = fopen("birthday.in", "r");
  if (!f)
    fail("Failed to open input file.");

  res = fscanf(f, "%d", &R);
  if (res != 1)
    fail("Failed to read R from input file.");

  res = fscanf(f, "%d", &N);
  if (res != 1)
    fail("Failed to read N from input file.");

  if (N < 1 || N > MAX_N)
    fail("N is out of bounds.");

  for (i = 0; i < N; i++) {
    res = fscanf(f, "%d%d%d%d", &X1[i], &Y1[i], &X2[i], &Y2[i]);
    if (res != 4)
      fail("Failed to read data from input file.");
  }
  fclose(f);

  int answer = pieces(R, N, X1, Y1, X2, Y2);

  printf("%d\n", answer);

  return 0;
}

File: ioi/2013/day0-1/birthday/grader.pas
program grader;

uses birthday;

procedure fail(s: string);
begin
    writeln(stderr, s);
    halt(1);
end;

var
    R, N, i: longint;
    X1, Y1, X2, Y2: array of Longint;
    f: text;
    answer: longint;

begin
    assign(f, 'birthday.in');
    {$I-} reset(f); {$I+}
    if IOResult <> 0 then
        fail('Failed to open input file.');

    readln(f, R);
    readln(f, N);

    setlength(X1, N);
    setlength(Y1, N);
    setlength(X2, N);
    setlength(Y2, N);

    for i := 0 to N-1 do
        readln(f, X1[i], Y1[i], X2[i], Y2[i]);
    close(f);

    answer := pieces(R, N, X1, Y1, X2, Y2);

    writeln(answer);
end.
